[CBRD-24155] Add a system parameter to set core count for thread worker pool

hgryoo opened the pull request:
http://jira.cubrid.org/browse/CBRD-24155

- a new parameter, "thread_core_count" is added
  - Change (Tune) the value entered by the user considering max_client and system core count
  - The behavior of partitioning workers according to the # of cores in the thread worker pool is changed to use "thread_core_count" as partitioning factor

please refer to the JIRA description for detailed implementation.

hgryoo committed:
add thread_core_count parameter

mhoh3963 commented on src/base/system_parameter.c:
```diff
@@ -2377,6 +2380,15 @@ static UINT64 prm_first_log_pageid_lower = 0LL;
 static UINT64 prm_first_log_pageid_upper = LOGPAGEID_MAX;
 static unsigned int prm_first_log_pageid_flag = 0;
 
+static const int lower_bound_core_count = (prm_css_max_clients_default / 3);
```
How about use prm_css_max_clients_lower in stead of prm_css_max_clients_default ?

hgryoo committed:
Merge remote-tracking branch 'upstream/develop' into CBRD-24155

hgryoo committed:
change thread_core_count prm's range and tuning rules

- set upper_count as 1024
- refactor max_workers, max_connections, max_task_count to be exposed
- refactor tuning thread_core_count

hgryoo committed:
revert including header

hgryoo commented on src/base/system_parameter.c:
```diff
@@ -10711,6 +10749,7 @@ prm_tune_parameters (void)
   SYSPRM_PARAM *ha_check_disk_failure_interval_prm;
   SYSPRM_PARAM *test_mode_prm;
   SYSPRM_PARAM *tz_leap_second_support_prm;
+  SYSPRM_PARAM *thread_core_count_prm;
```
```suggestion
```

hgryoo committed:
remove unused line
